crates/sui-types/src/effects/object_change.rs

```diff
@@ -125,3 +205,287 @@ pub enum ObjectOut {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Split,
+                value: AccumulatorValue::Integer(50),
+            },
+        ];
+
+        let result = AccumulatorWriteV1::merge(writes);
+        assert_eq!(result.operation, AccumulatorOperation::Split);
+        assert_eq!(result.value, AccumulatorValue::Integer(350));
+    }
+
+    #[test]
+    fn test_merge_event_digests_single() {
+        let addr = test_accumulator_address();
+        let digest1 = Digest::random();
+        let digest2 = Digest::random();
+
+        let write = AccumulatorWriteV1 {
+            address: addr.clone(),
+            operation: AccumulatorOperation::Merge,
+            value: AccumulatorValue::EventDigest(vec![(0, digest1), (1, digest2)]),
+        };
+
+        let result = AccumulatorWriteV1::merge(vec![write.clone()]);
+        assert_eq!(result, write);
+    }
+
+    #[test]
+    fn test_merge_event_digests_multiple() {
+        let addr = test_accumulator_address();
+        let digest1 = Digest::random();
+        let digest2 = Digest::random();
+        let digest3 = Digest::random();
+        let digest4 = Digest::random();
+
+        let writes = vec![
+            AccumulatorWriteV1 {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Merge,
+                value: AccumulatorValue::EventDigest(vec![(0, digest1), (1, digest2)]),
+            },
+            AccumulatorWriteV1 {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Merge,
+                value: AccumulatorValue::EventDigest(vec![(2, digest3)]),
+            },
+            AccumulatorWriteV1 {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Merge,
+                value: AccumulatorValue::EventDigest(vec![(3, digest4)]),
+            },
+        ];
+
+        let result = AccumulatorWriteV1::merge(writes);
+        assert_eq!(result.operation, AccumulatorOperation::Merge);
+        if let AccumulatorValue::EventDigest(digests) = result.value {
+            assert_eq!(digests.len(), 4);
+            assert_eq!(digests[0], (0, digest1));
+            assert_eq!(digests[1], (1, digest2));
+            assert_eq!(digests[2], (2, digest3));
+            assert_eq!(digests[3], (3, digest4));
+        } else {
+            panic!("Expected EventDigest value");
+        }
+    }
+
+    #[test]
+    fn test_merge_event_digests_empty_list() {
+        let addr = test_accumulator_address();
+        let writes = vec![
+            AccumulatorWriteV1 {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Merge,
+                value: AccumulatorValue::EventDigest(vec![]),
+            },
+            AccumulatorWriteV1 {
+                address: addr.clone(),
+                operation: AccumulatorOperation::Merge,
+                value: AccumulatorValue::EventDigest(vec![]),
```

Comment:
is it even possible to have an empty list of events? I would think not. We could use `NonEmpty<T>` instead of `Vec<T>` in this case